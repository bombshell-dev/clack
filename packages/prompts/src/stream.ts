import { stripVTControlCharacters as strip } from 'node:util';
import color from 'picocolors';
import { S_BAR, S_ERROR, S_INFO, S_STEP_SUBMIT, S_SUCCESS, S_WARN } from './common.js';
import type { LogMessageOptions } from './log.js';

const prefix = `${color.gray(S_BAR)}  `;

// TODO (43081j): this currently doesn't support custom `output` writables
// because we rely on `columns` existing (i.e. `process.stdout.columns).
//
// If we want to support `output` being passed in, we will need to use
// a condition like `if (output insance Writable)` to check if it has columns
export const stream = {
	message: async (
		iterable: Iterable<string> | AsyncIterable<string>,
		{ symbol = color.gray(S_BAR) }: LogMessageOptions = {}
	) => {
		process.stdout.write(`${color.gray(S_BAR)}\n${symbol}  `);
		let lineWidth = 3;
		for await (let chunk of iterable) {
			chunk = chunk.replace(/\n/g, `\n${prefix}`);
			if (chunk.includes('\n')) {
				lineWidth = 3 + strip(chunk.slice(chunk.lastIndexOf('\n'))).length;
			}
			const chunkLen = strip(chunk).length;
			if (lineWidth + chunkLen < process.stdout.columns) {
				lineWidth += chunkLen;
				process.stdout.write(chunk);
			} else {
				process.stdout.write(`\n${prefix}${chunk.trimStart()}`);
				lineWidth = 3 + strip(chunk.trimStart()).length;
			}
		}
		process.stdout.write('\n');
	},
	info: (iterable: Iterable<string> | AsyncIterable<string>) => {
		return stream.message(iterable, { symbol: color.blue(S_INFO) });
	},
	success: (iterable: Iterable<string> | AsyncIterable<string>) => {
		return stream.message(iterable, { symbol: color.green(S_SUCCESS) });
	},
	step: (iterable: Iterable<string> | AsyncIterable<string>) => {
		return stream.message(iterable, { symbol: color.green(S_STEP_SUBMIT) });
	},
	warn: (iterable: Iterable<string> | AsyncIterable<string>) => {
		return stream.message(iterable, { symbol: color.yellow(S_WARN) });
	},
	/** alias for `log.warn()`. */
	warning: (iterable: Iterable<string> | AsyncIterable<string>) => {
		return stream.warn(iterable);
	},
	error: (iterable: Iterable<string> | AsyncIterable<string>) => {
		return stream.message(iterable, { symbol: color.red(S_ERROR) });
	},
};
