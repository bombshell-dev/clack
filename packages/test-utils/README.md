# `@clack/test-utils`

A bunch of useful test utiltiies for use inside clack.
