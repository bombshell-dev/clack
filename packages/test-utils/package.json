{
  "name": "@clack/test-utils",
  "private": true,
  "version": "1.0.0-alpha.1",
  "type": "module",
  "main": "./dist/index.js",
  "module": "./dist/index.js",
  "exports": {
    ".": {
      "types": "./dist/index.d.ts",
      "default": "./dist/index.js"
    },
    "./package.json": "./package.json"
  },
  "types": "./dist/index.d.ts",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/bombshell-dev/clack.git",
    "directory": "packages/test-utils"
  },
  "bugs": {
    "url": "https://github.com/bombshell-dev/clack/issues"
  },
  "homepage": "https://github.com/bombshell-dev/clack/tree/main/packages/test-utils#readme",
  "files": ["dist", "CHANGELOG.md"],
  "author": {
    "name": "James Garbutt",
    "url": "https://github.com/43081j"
  },
  "license": "MIT",
  "packageManager": "pnpm@9.14.2",
  "scripts": {
    "build": "tsc",
    "prepack": "pnpm build"
  },
  "dependencies": {
  },
  "devDependencies": {
  }
}
