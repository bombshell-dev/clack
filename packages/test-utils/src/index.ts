export { MockReadable } from './mock-readable.js';
export { MockWritable } from './mock-writable.js';

export function nextTick(): Promise<void> {
	return new Promise<void>((resolve) => {
		setTimeout(resolve, 0);
	});
}
