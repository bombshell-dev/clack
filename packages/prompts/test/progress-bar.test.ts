import process from 'node:process';
import { EventEmitter } from 'node:stream';
import { afterAll, afterEach, beforeAll, beforeEach, describe, expect, test, vi } from 'vitest';
import * as prompts from '../src/index.js';
import type { ProgressOptions } from '../src/index.js';
import { MockWritable } from './test-utils.js';

describe.each(['true', 'false'])('prompts - progress (isCI = %s)', (isCI) => {
	let originalCI: string | undefined;
	let output: MockWritable;

	beforeAll(() => {
		originalCI = process.env.CI;
		process.env.CI = isCI;
	});

	afterAll(() => {
		process.env.CI = originalCI;
	});

	beforeEach(() => {
		output = new MockWritable();
		vi.useFakeTimers();
	});

	afterEach(() => {
		vi.restoreAllMocks();
		vi.useRealTimers();
	});

	test('returns progress API', () => {
		const api = prompts.progress({ output });

		expect(api.stop).toBeTypeOf('function');
		expect(api.start).toBeTypeOf('function');
		expect(api.message).toBeTypeOf('function');
		expect(api.advance).toBeTypeOf('function');
	});

	describe('start', () => {
		test('renders frames at interval', () => {
			const result = prompts.progress({ output });

			result.start();

			// there are 4 frames
			for (let i = 0; i < 4; i++) {
				vi.advanceTimersByTime(80);
			}

			expect(output.buffer).toMatchSnapshot();
		});

		test('renders message', () => {
			const result = prompts.progress({ output });

			result.start('foo');

			vi.advanceTimersByTime(80);

			expect(output.buffer).toMatchSnapshot();
		});

		test('renders timer when indicator is "timer"', () => {
			const result = prompts.progress({ output, indicator: 'timer' });

			result.start();

			vi.advanceTimersByTime(80);

			expect(output.buffer).toMatchSnapshot();
		});
	});

	describe('stop', () => {
		test('renders submit symbol and stops progress', () => {
			const result = prompts.progress({ output });

			result.start();

			vi.advanceTimersByTime(80);

			result.stop();

			vi.advanceTimersByTime(80);

			expect(output.buffer).toMatchSnapshot();
		});

		test('renders cancel symbol if code = 1', () => {
			const result = prompts.progress({ output });

			result.start();

			vi.advanceTimersByTime(80);

			result.stop('', 1);

			expect(output.buffer).toMatchSnapshot();
		});

		test('renders error symbol if code > 1', () => {
			const result = prompts.progress({ output });

			result.start();

			vi.advanceTimersByTime(80);

			result.stop('', 2);

			expect(output.buffer).toMatchSnapshot();
		});

		test('renders message', () => {
			const result = prompts.progress({ output });

			result.start();

			vi.advanceTimersByTime(80);

			result.stop('foo');

			expect(output.buffer).toMatchSnapshot();
		});

		test('renders message without removing dots', () => {
			const result = prompts.progress({ output });

			result.start();

			vi.advanceTimersByTime(80);

			result.stop('foo.');

			expect(output.buffer).toMatchSnapshot();
		});
	});

	describe('message', () => {
		test('sets message for next frame', () => {
			const result = prompts.progress({ output });

			result.start();

			vi.advanceTimersByTime(80);

			result.message('foo');

			vi.advanceTimersByTime(80);

			expect(output.buffer).toMatchSnapshot();
		});
	});

	describe('process exit handling', () => {
		let processEmitter: EventEmitter;

		beforeEach(() => {
			processEmitter = new EventEmitter();

			// Spy on process methods
			vi.spyOn(process, 'on').mockImplementation((ev, listener) => {
				processEmitter.on(ev, listener);
				return process;
			});
			vi.spyOn(process, 'removeListener').mockImplementation((ev, listener) => {
				processEmitter.removeListener(ev, listener);
				return process;
			});
		});

		afterEach(() => {
			processEmitter.removeAllListeners();
		});

		test('uses default cancel message', () => {
			const result = prompts.progress({ output });
			result.start('Test operation');

			processEmitter.emit('SIGINT');

			expect(output.buffer).toMatchSnapshot();
		});

		test('uses custom cancel message when provided directly', () => {
			const result = prompts.progress({
				output,
				cancelMessage: 'Custom cancel message',
			});
			result.start('Test operation');

			processEmitter.emit('SIGINT');

			expect(output.buffer).toMatchSnapshot();
		});

		test('uses custom error message when provided directly', () => {
			const result = prompts.progress({
				output,
				errorMessage: 'Custom error message',
			});
			result.start('Test operation');

			processEmitter.emit('exit', 2);

			expect(output.buffer).toMatchSnapshot();
		});

		test('uses global custom cancel message from settings', () => {
			// Store original message
			const originalCancelMessage = prompts.settings.messages.cancel;
			// Set custom message
			prompts.settings.messages.cancel = 'Global cancel message';

			const result = prompts.progress({ output });
			result.start('Test operation');

			processEmitter.emit('SIGINT');

			expect(output.buffer).toMatchSnapshot();

			// Reset to original
			prompts.settings.messages.cancel = originalCancelMessage;
		});

		test('uses global custom error message from settings', () => {
			// Store original message
			const originalErrorMessage = prompts.settings.messages.error;
			// Set custom message
			prompts.settings.messages.error = 'Global error message';

			const result = prompts.progress({ output });
			result.start('Test operation');

			processEmitter.emit('exit', 2);

			expect(output.buffer).toMatchSnapshot();

			// Reset to original
			prompts.settings.messages.error = originalErrorMessage;
		});

		test('prioritizes direct options over global settings', () => {
			// Store original messages
			const originalCancelMessage = prompts.settings.messages.cancel;
			const originalErrorMessage = prompts.settings.messages.error;

			// Set custom global messages
			prompts.settings.messages.cancel = 'Global cancel message';
			prompts.settings.messages.error = 'Global error message';

			const result = prompts.progress({
				output,
				cancelMessage: 'Progress cancel message',
				errorMessage: 'Progress error message',
			});
			result.start('Test operation');

			processEmitter.emit('SIGINT');
			expect(output.buffer).toMatchSnapshot();

			// Reset buffer
			output.buffer = [];

			const result2 = prompts.progress({
				output,
				cancelMessage: 'Progress cancel message',
				errorMessage: 'Progress error message',
			});
			result2.start('Test operation');

			processEmitter.emit('exit', 2);
			expect(output.buffer).toMatchSnapshot();

			// Reset to original values
			prompts.settings.messages.cancel = originalCancelMessage;
			prompts.settings.messages.error = originalErrorMessage;
		});
	});

	describe('style', () => {
		test.each(['block', 'heavy', 'light'] satisfies Array<ProgressOptions['style']>)(
			'renders %s progressbar',
			(style) => {
				const result = prompts.progress({ output, style, max: 2, size: 10 });
				result.start();
				vi.advanceTimersByTime(160);
				result.advance();
				vi.advanceTimersByTime(160);
				result.stop();

				expect(output.buffer).toMatchSnapshot();
			}
		);
	});
});
