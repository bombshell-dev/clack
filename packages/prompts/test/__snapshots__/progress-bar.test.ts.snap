// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`prompts - progress (isCI = false) > message > sets message for next frame 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m foo",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > prioritizes cancel option over global setting 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Progress cancel message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > prioritizes error option over global setting 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m▲[39m  Progress error message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > uses custom cancel message when provided directly 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Custom cancel message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > uses custom error message when provided directly 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m▲[39m  Custom error message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > uses default cancel message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Canceled
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > uses global custom cancel message from settings 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Global cancel message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > process exit handling > uses global custom error message from settings 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m▲[39m  Global error message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > start > renders frames at interval 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◓[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◑[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
]
`;

exports[`prompts - progress (isCI = false) > start > renders message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m foo",
]
`;

exports[`prompts - progress (isCI = false) > start > renders timer when indicator is "timer" 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m  [0s]",
]
`;

exports[`prompts - progress (isCI = false) > stop > renders cancel symbol if code = 1 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[31m■[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > stop > renders error symbol if code > 1 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[31m▲[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > stop > renders message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  foo
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > stop > renders message without removing dots 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  foo.
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > stop > renders submit symbol and stops progress 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > style > renders block progressbar 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m██████████[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m[39m[2m██████████[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◓[39m  [35m█████[39m[2m█████[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◑[39m  [35m█████[39m[2m█████[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > style > renders heavy progressbar 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m[39m[2m━━━━━━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◓[39m  [35m━━━━━[39m[2m━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◑[39m  [35m━━━━━[39m[2m━━━━━[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = false) > style > renders light progressbar 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m──────────[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m[39m[2m──────────[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◓[39m  [35m─────[39m[2m─────[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◑[39m  [35m─────[39m[2m─────[22m ",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > message > sets message for next frame 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m foo...",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > prioritizes cancel option over global setting 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Progress cancel message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > prioritizes error option over global setting 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m▲[39m  Progress error message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > uses custom cancel message when provided directly 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Custom cancel message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > uses custom error message when provided directly 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m▲[39m  Custom error message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > uses default cancel message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Canceled
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > uses global custom cancel message from settings 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m■[39m  Global cancel message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > process exit handling > uses global custom error message from settings 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[31m▲[39m  Global error message
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > start > renders frames at interval 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
]
`;

exports[`prompts - progress (isCI = true) > start > renders message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m foo...",
]
`;

exports[`prompts - progress (isCI = true) > start > renders timer when indicator is "timer" 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
]
`;

exports[`prompts - progress (isCI = true) > stop > renders cancel symbol if code = 1 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[31m■[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > stop > renders error symbol if code > 1 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[31m▲[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > stop > renders message 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  foo
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > stop > renders message without removing dots 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  foo.
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > stop > renders submit symbol and stops progress 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > style > renders block progressbar 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m██████████[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m█████[39m[2m█████[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > style > renders heavy progressbar 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m━━━━━━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m━━━━━[39m[2m━━━━━[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;

exports[`prompts - progress (isCI = true) > style > renders light progressbar 1`] = `
[
  "<cursor.hide>",
  "[90m│[39m
",
  "[35m◒[39m  [35m[39m[2m──────────[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[35m◐[39m  [35m─────[39m[2m─────[22m ...",
  "
",
  "<cursor.backward count=999>",
  "<erase.down>",
  "[32m◇[39m  
",
  "<cursor.show>",
]
`;
